http/beaconblockproposal.go

```diff
@@ -169,8 +185,12 @@ func (s *Service) beaconBlockProposalV1(ctx context.Context, slot phase0.Slot, r
 	if resp.Data.Slot != slot {
 		return nil, errors.New("beacon block proposal not for requested slot")
 	}
-	if !bytes.Equal(resp.Data.Body.RANDAOReveal[:], randaoReveal[:]) {
-		return nil, errors.New("beacon block proposal has incorrect RANDAO reveal")
+	// Only check the RANDAO reveal if we are not connected to DVT middleware, as that
+	// will return a full signature rather than the element signature we passed it.
+	if !s.connectedToDVTMiddleware {
+		if !bytes.Equal(resp.Data.Body.RANDAOReveal[:], randaoReveal[:]) {
+			return nil, errors.New("beacon block proposal has incorrect RANDAO reveal")
+		}
 	}
 	if !bytes.Equal(resp.Data.Body.Graffiti[:], graffiti) {
```

Comment:
can we also include this graffiti check inside the above if block?

Reply:
Done.